### PR TITLE
Allowing ability to add custom env into pgbouncer container

### PR DESCRIPTION
**Pull request by amoghrajesh**
Currently, we do not allow adding custom environment variables into the pgbouncer container. It should be allowed when we want to set some custom configs through env.

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
spelling/docs issue with pgbouncer (hint: put it in `` to avoid it).

**Comment by amoghrajesh** [comment]
Thanks for the tip, @potiuk
Don't think I would've ever gotten it otherwise. Is it documented anywhere?

File: chart/templates/pgbouncer/pgbouncer-deployment.yaml [hunk 1]
```diff
@@ -101,6 +101,13 @@ spec:
           args: {{ tpl (toYaml .Values.pgbouncer.args) . | nindent 12 }}
           {{- end }}
           resources: {{- toYaml .Values.pgbouncer.resources | nindent 12 }}
+          {{- if .Values.pgbouncer.env }}
+          env:
+            {{- range $i, $config := .Values.pgbouncer.env }}
+            - value: {{ $config.value }}
+              name: {{ $config.name }}
+            {{- end }}
+          {{- end }}
```
Comment (hussein-awala) [inline chart/templates/pgbouncer/pgbouncer-deployment.yaml]: Could we add the value as it is instead of looping over the values?

Comment (amoghrajesh) [reply chart/templates/pgbouncer/pgbouncer-deployment.yaml]: .env will be an array of key value pairs. How do you propose we add it instead?

Comment (hussein-awala) [inline chart/templates/pgbouncer/pgbouncer-deployment.yaml]: Just like that:
```suggestion
          {{- with .Values.pgbouncer.env }}
          env: {{- toYaml . | nindent 12 }}
          {{- end }}
```

**Review by hussein-awala — APPROVED** [review]

**Pull request merged**